buildSrc/src/main/groovy/org/logstash/gradle/tooling/StackVersionSelector.groovy

```diff
@@ -0,0 +1,148 @@
+package org.logstash.gradle.tooling
+
+import org.gradle.api.GradleException
+import groovy.json.JsonSlurper
+
+class StackVersionSelector {
+
+    private final String artifactVersionsApi
+
+    public StackVersionSelector(String artifactVersionsApi) {
+        this.artifactVersionsApi = artifactVersionsApi
+    }
+
+    def selectClosestVersion(String version) {
+        String apiResponse = artifactVersionsApi.toURL().text
+        def dlVersions = new JsonSlurper().parseText(apiResponse)
+        List<StackVersion> versions = dlVersions['versions'].collect {s -> StackVersion.asVersion(s)}
+
+        def versionQualifier = System.getenv('VERSION_QUALIFIER')
+        def isReleaseBuild = System.getenv('RELEASE') == "1" || versionQualifier
+
+        String qualifiedVersion = selectClosestInList(StackVersion.asVersion(isReleaseBuild ? version : "${version}-SNAPSHOT"), versions)
+        if (qualifiedVersion == null) {
+            throw new GradleException("could not find the current artifact from the artifact-api ${artifactVersionsApi} for ${version}")
```

Comment:
"the current" artifact not a good way to describe an artifact, we should specific why one it is. e.g.:


cloud not find **<name-of-artifact>** version ${version} in the artifact-api (${artifactVersionApi})

So I guess it would read:

could not find artifact Elasticsearch 8.0.0-alpha1 in the arfifact-api (https://xzy)/

or something like:

could not find Elasticsearch (elasticsearch-8.0.0-alpha1) in the artifact api (https://xxxx)

Reply:
Good point, I would change the error to:
```
could not find Elastic Stack version (8.0.0-alpha1) in the artifact api (https://xxxx)
```
because at this point the `StackVersionSelector` is only used to select the version of the global stack and not for a specific artifact, the `StackVersionSelector` hasn't got that information